### ISSUE TITLE
InputMode never changes

### ISSUE THREAD

**Issue by ookiegajwa** [GeyserMC/Floodgate#430]
`FloodgatePlayer#getInputMode`, which should give the current input method being used by the player, is always the same: `KEYBOARD_MOUSE` for Android devices, and `UNKNOWN` for iOS and Windows (and I haven't tested console).

I'm not sure if this is a bug in Bedrock Edition itself, or if something in Geyser/Floodgate is causing this. I feel like this used to work better, because I used to have some logic in my server that used `InputMode`, but one day it broke.

It would be great if someone could get this to work again, but if it is a Minecraft bug then I'm not sure we can.

**Comment by onebeastchris** [comment]
Are you using the Floodgate API in a proxy setup, e.g. behind a velocity/bungee proxy?

**Comment by ookiegajwa** [comment]
No, it is installed directly on the server

**Comment by onebeastchris** [comment]
Mind sending a Geyser dump?
Run `geyser dump` in console, and send the link you get here

**Comment by ookiegajwa** [comment]
https://dump.geysermc.org/WlIJlq4HBue7heMVWsfJKZmAQmdCxw3G

Here you are :)

**Comment by onebeastchris** [comment]
Any errors in the server log? Additionally, does it just affect 1.20.10 bedrock players, or all versions?

**Comment by ookiegajwa** [comment]
No errors. I have been having issues with this for a while now, and the issue was most recently reported to me by a 1.20.0 player, so probably all versions.

**Comment by ookiegajwa** [comment]
`DeviceOs` does work, so some device info is definitely getting sent to the server.

**Comment by Tim203** [comment]
Is this still an issue in the latest version?

**Comment by ookiegajwa** [comment]
Yes - now I get `UNKNOWN` from all devices I have access to (Android emulator, iOS native, Windows 11 native)

**Comment by Tim203** [comment]
Do you still have a setup where this issue is occuring?

**Comment by ookiegajwa** [comment]
Yes, I still receive `UNKNOWN` input modes from iOS native (can check Android emulator and Windows 11 later if necessary).

**Comment by Tim203** [comment]
Are you in our discord server? And are you using a plugin that shows unknown or are you using the api yourself?

**Comment by ookiegajwa** [comment]
Yes I am, and I'm using the API myself.

**Comment by Tim203** [comment]
What is your Discord username? I had trouble finding you
